vault/identity_store_oidc.go

```diff
@@ -609,11 +699,12 @@ func (i *IdentityStore) pathOIDCGenerateToken(ctx context.Context, req *logical.
 
 	now := time.Now()
 	idToken := idToken{
-		Issuer:   config.effectiveIssuer,
-		Subject:  req.EntityID,
-		Audience: role.ClientID,
-		Expiry:   now.Add(role.TokenTTL).Unix(),
-		IssuedAt: now.Unix(),
+		Issuer:    config.effectiveIssuer,
+		Namespace: strings.TrimSuffix(ns.Path, "/"),
```

Comment:
Should we be using namespace ID here instead?

Reply:
Good question. Is the human-readable string useful/convenient, or exposing information unnecessarily?  I'd say the latter and would vote for changing it.

Reply:
I'm gonna err on the side of ID for now.